### PR TITLE
Rewrite lambdas to method references for instance methods

### PR DESCRIPTION
**Pull request by ferozco**
## Before this PR
We would only re-write lambdas with zero parameters that invoked static methods to method references

## After this PR
==COMMIT_MSG==
Rewrite single parameter lambdas that only invoke an zero param instance method to a method reference
==COMMIT_MSG==

## Possible downsides?
N/A

### CODE REVIEW

**Comment by carterkozak** [comment]
Nice, we should test this on a large internal project before merging. I found a lot of bugs in the original implementation that way!

File: baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LambdaMethodReference.java
Comment (carterkozak) [inline baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/LambdaMethodReference.java]:
```suggestion
        return !root.getParameters().isEmpty();
```

**Review by carterkozak — APPROVED** [review]
Approving pending tests in a large internal project

**Comment by carterkozak** [comment]
Everything is green.

**Comment by carterkozak** [comment]
Needs an approval from you as well since my commits merged into this branch :-)

**Comment by ferozco** [comment]
woohooo 👍

**Comment by ferozco** [comment]
🤖

**Pull request merged**
